casepro/msgs/views.py

```diff
@@ -434,3 +454,178 @@ def post(self, request, *args, **kwargs):
     class Read(BaseDownloadView):
         title = _("Download Replies")
         filename = 'reply_export.xls'
+
+
+class FaqSearchMixin(object):
+    def derive_search(self):
+        """
+        Collects and prepares FAQ search parameters into JSON serializable dict
+        """
+        label = self.request.GET.get('label', None)
+        text = self.request.GET.get('text', None)
+        language = self.request.GET.get('language', None)
+
+        return {
+            'label': label,
+            'text': text,
+            'language': language,
+        }
+
+
+class FaqCRUDL(SmartCRUDL):
+    model = FAQ
+    actions = ('list', 'create', 'read', 'update', 'delete', 'search', 'import', 'languages')
+
+    class List(OrgPermsMixin, SmartListView):
+        fields = ('question', 'answer', 'language', 'parent')
+        default_order = ('-parent', 'question')
+
+        def derive_queryset(self, **kwargs):
+            return FAQ.get_all(self.request.org)
+
+        def get_context_data(self, **kwargs):
+            context = super(FaqCRUDL.List, self).get_context_data(**kwargs)
+            # change the language code to full name for display only
+            for faq in context['object_list']:
+                faq.language = get_language_name(faq.language)
```

Comment:
If that's awkward to use in templates, we can switch both of those to return a non-model class, something like 

```python
class Language(object):
  def __init__(self, code):
    self.code = code
    self.name = get_language_name(code)

  def as_json():
    return {'code': self.language, 'name': self.name}